---
title: "Infer ASVs with DADA2"
output: html_document
date: "`r Sys.Date()`"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      fig.align = "center",
                      fig.path = "../figures/01_DADA2/") # send any figure output to this folder
```

#Goals of this file

1. Use raw fastq files and generate quality plots to assess quality of reads.
2. Filter out and trim outbad sequences and bases from our sequencing files.
3. Write out fastq files with high quality sequences.
4. Evaluate the quality from our filter and trim.

#Load Libraries
```{r load-libraries}
#install.packages("devtools")
library(devtools)

#devtools::install_github("benjjneb/dada2")

library(dada2)
```



# Load data 
```{r load-data}
#Set the raw fastq path to the raw sequencing files
#Path to the fastq files

raw_fastqs_path <- "data/01_DADA2/01_raw_gzipped_fastqs"
raw_fastqs_path

#What files are in this path?
list.files(raw_fastqs_path)

#How many files are there?
str(list.files(raw_fastqs_path))

# Create a vector of forward reads
forward_reads <- list.files(raw_fastqs_path, pattern = "R1_001.fastq.gz", full.names = TRUE)
#Intuition Check
head(forward_reads)

# Create a vector of reverse reads

reverse_reads <- list.files(raw_fastqs_path, pattern = "R2_001.fastq.gz", full.names = TRUE)

head(reverse_reads)
```

#Quality Plots
```{r raw-quality-plot}

#Randomly select two samples from dataset to evaluate

random_samples <- sample(1:length(reverse_reads),size = 2)
random_samples

#Calculate and plot quality of these two samples

plotQualityProfile(forward_reads[random_samples])
plotQualityProfile(reverse_reads[random_samples])
```

