---
title: "Infer ASVs with DADA2"
output: html_document
date: "`r Sys.Date()`"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      fig.align = "center",
                      fig.path = "../figures/01_DADA2/") # send any figure output to this folder
```

```{r set-seed}

#Any number can be chosen
set.seed(238428)


```

#Goals of this file

1. Use raw fastq files and generate quality plots to assess quality of reads.
2. Filter out and trim outbad sequences and bases from our sequencing files.
3. Write out fastq files with high quality sequences.
4. Evaluate the quality from our filter and trim.

#Load Libraries
```{r load-libraries}
#install.packages("devtools")
library(devtools)

#devtools::install_github("benjjneb/dada2")

library(dada2)
```



# Load data 
```{r load-data}
#Set the raw fastq path to the raw sequencing files
#Path to the fastq files

raw_fastqs_path <- "data/01_DADA2/01_raw_gzipped_fastqs"
raw_fastqs_path

#What files are in this path?
list.files(raw_fastqs_path)

#How many files are there?
str(list.files(raw_fastqs_path))

# Create a vector of forward reads
forward_reads <- list.files(raw_fastqs_path, pattern = "R1_001.fastq.gz", full.names = TRUE)
#Intuition Check
head(forward_reads)

# Create a vector of reverse reads

reverse_reads <- list.files(raw_fastqs_path, pattern = "R2_001.fastq.gz", full.names = TRUE)

head(reverse_reads)
```

#Quality Plots
```{r raw-quality-plot}

#Randomly select two samples from dataset to evaluate

random_samples <- sample(1:length(reverse_reads),size = 2)
random_samples

#Calculate and plot quality of these two samples

plotQualityProfile(forward_reads[random_samples])
plotQualityProfile(reverse_reads[random_samples])
```

#Prepare a placeholder for filtered reads 
```{r prep-filtered-sequences}
#Place filtered reads into filtered_fastqs_path

#Vector of our samples, extract the sample name from our files

samples <- sapply(strsplit(basename(forward_reads),"_"), `[`,1)
head(samples)

filtered_fastqs_path <- "data/01_DADA2/02_filtered_fastqs"

#Create two variables: filtered_F, filtered_R
filtered_forward_reads <- file.path(filtered_fastqs_path, paste0(samples, "_R1_filtered.fastq.gz"))
length(filtered_forward_reads)

filtered_reverse_reads <- file.path(filtered_fastqs_path, paste0(samples, "_R2_filtered.fastq.gz"))
head(filtered_reverse_reads)
length(filtered_reverse_reads)
```



# Filter and trim reads
# Parameters of filter and trim **DEPEND ON THE DATASET**
# maxN = number of N bases. Remove all N's from the data 
# maxEE = Quality filtering threshold applied to expected errors. Here if there are two expected errors, it's okay. But more than 2, throw away the sequence. Two values: first for forward reads, second for reverse reads
# trimLeft = Number of nucleotides to remove from the start of each read. We will be removing the first three bases


```{r filter-and-trim}
#Assign a vector to filtered reads
#Trim out first 3 bases
#Write out filtered fastq files
filtered_reads <-  filterAndTrim(fwd = forward_reads, filt = filtered_forward_reads, rev = reverse_reads, filt.rev = filtered_reverse_reads, maxN = 0, maxEE = c(2,2), trimLeft = 3, truncQ = 2, rm.phix = TRUE, compress = TRUE)



```


# Session Information
```{r session-info}
# Ensure reproducibility

devtools::session_info()

```
